server/modules/sender.js

```diff
@@ -10,20 +11,24 @@ const COMMAND_TYPE = 'Command';
 const QUERY_TYPE = 'Query';
 
 module.exports = {
-  sendCommand(parameters, callback) {
+  sendCommand(handler, parameters, callback) {
+    assert(typeof handler === 'function');
+    assert(typeof parameters === 'object' && parameters !== null);
```

Comment:
`typeof [] // 'object'`
If we are happy with that, that's fine. If not, we might want to add an `!Array.isArray(parameters)`